pulsar-client-kafka-compat/src/main/java/org/apache/kafka/clients/consumer/PulsarKafkaConsumer.java

```diff
@@ -0,0 +1,482 @@
+                    conf.setConsumerName(ConsumerName.generateRandomName());
+                    for (int i = 0; i < partitionMetadata.partitions; i++) {
+                        String partitionName = DestinationName.get(topic).getPartition(i).toString();
+                        org.apache.pulsar.client.api.Consumer consumer = client.subscribe(partitionName, groupId, conf);
+                        consumers.put(partitionName, consumer);
+                    }
+                } else {
+                    // Topic has a single partition
+                    org.apache.pulsar.client.api.Consumer consumer = client.subscribe(topic, groupId, conf);
+                    consumers.put(topic, consumer);
+                }
+            } catch (InterruptedException | ExecutionException | PulsarClientException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Override
+    public void subscribe(Collection<String> topics, ConsumerRebalanceListener callback) {
+        throw new UnsupportedOperationException("ConsumerRebalanceListener is not supported");
+    }
+
+    @Override
+    public void assign(Collection<TopicPartition> partitions) {
+        throw new UnsupportedOperationException("Cannot manually assign partitions");
+    }
+
+    @Override
+    public void subscribe(Pattern pattern, ConsumerRebalanceListener callback) {
+        throw new UnsupportedOperationException("Cannot subscribe with topic name pattern");
+    }
+
+    @Override
+    public void unsubscribe() {
+        consumers.values().forEach(c -> {
+            try {
+                c.unsubscribe();
+            } catch (PulsarClientException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public ConsumerRecords<K, V> poll(long timeoutMillis) {
+        try {
+            QueueItem item = receivedMessages.poll(timeoutMillis, TimeUnit.MILLISECONDS);
+            if (item == null) {
+                return (ConsumerRecords<K, V>) ConsumerRecords.EMPTY;
+            }
+
+            if (isAutoCommit) {
+                // Commit the offset of previously dequeued messages
+                commitAsync();
+            }
+
+            String topic = item.consumer.getTopic();
+            Message msg = item.message;
+            MessageIdImpl msgId = (MessageIdImpl) msg.getMessageId();
+            long offset = getOffset(msgId);
+
+            int partition = msgId.getPartitionIndex();
+
+            TopicPartition tp = new TopicPartition(topic, partition);
+
+            K key = getKey(topic, msg);
+            V value = valueDeserializer.deserialize(topic, msg.getData());
+
+            ConsumerRecord<K, V> consumerRecord = new ConsumerRecord<>(topic, partition, offset, msg.getPublishTime(),
+                    TimestampType.CREATE_TIME, -1, msg.hasKey() ? msg.getKey().length() : 0, msg.getData().length, key,
```

Comment:
we can support EVENT_TIME as well?

Reply:
Yes, fixed that on both producer and consumer, to pass and retrieve the event time.